src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -29,8 +29,37 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
     MTRDeviceStateUnreachable = 2
 };
 
+/**
+ * This enum is used to specify the type of logs requested from this device.
+ *
+ * The logs types are : End User Support, Network Diagnostics and Crash logs.
+ */
+typedef NS_ENUM(NSUInteger, MTRDiagnosticLogType) {
```

Comment:
Generally the default base type for NS_ENUM is (signed) `NSInteger`

Reply:
I just followed the example in this file . see MTRDeviceState. no specific preference here.

Reply:
done.